app/Launcher/Main.hs

```diff
@@ -36,9 +36,9 @@ import           Data.X509.Extra (failIfReasons, genRSA256KeyPair,
 -- Main
 --------------------------------------------------------------------------------
 
-main :: IO ()
+main :: IO ExitCode
```

Comment:
Errr, I really don't think that is right. The type of `main` should remain the same and you can then use `exitSuccess` or `exitFailure` to the the exectuable's exit code.